controllers/rabbitmqcluster_controller.go

```diff
@@ -160,6 +161,14 @@ func (r *RabbitmqClusterReconciler) Reconcile(ctx context.Context, req ctrl.Requ
 			return ctrl.Result{}, err
 		}
 
+		if err = r.reconcilePVC(ctx, builder, rabbitmqCluster, resource); err != nil {
+			rabbitmqCluster.Status.SetCondition(status.ReconcileSuccess, corev1.ConditionFalse, "FailedReconcilePVC", err.Error())
+			if statusErr := r.Status().Update(ctx, rabbitmqCluster); statusErr != nil {
+				logger.Error(statusErr, "Failed to update ReconcileSuccess condition state")
+			}
+			return ctrl.Result{}, err
+		}
+
```

Comment:
I see what you mean for sure. Personally I want to keep the logic in the helpers so that our `Reconcile()` stays as lean as possible. Particularly the type switch statement is verbose and adds too much implementation details to the `Reconcile()`

I refactored it differently from what you suggested here. Added `UpdateMayRequireStsRecreate()` to the builders interface and the logic is offloaded to each ResourceBuilder, and we no longer need a type switch statement here. It may seem to be an overkill, but I found it cleaner this way :) let me know what you think!

Reply:
Sounds good, we've used this pattern in the past and it worked well 👍